core/src/main/java/org/elasticsearch/indices/recovery/StartRecoveryRequest.java

```diff
@@ -94,6 +112,11 @@ public void readFrom(StreamInput in) throws IOException {
         targetNode = new DiscoveryNode(in);
         metadataSnapshot = new Store.MetadataSnapshot(in);
         primaryRelocation = in.readBoolean();
+        if (in.getVersion().onOrAfter(Version.V_6_0_0_alpha1_UNRELEASED)) {
+            startingSeqNo = in.readLong();
+        } else {
+            startingSeqNo = SequenceNumbersService.UNASSIGNED_SEQ_NO;
```

Comment:
This has to be reverted due to test infrastructure that randomized stream versions. I pushed 6a0b70e60b45298f380149ac415a27ea0dac7e42.